build/build.go

```diff
@@ -1235,6 +1238,33 @@ func LoadInputs(ctx context.Context, d driver.Driver, inp Inputs, pw progress.Wr
 			target.FrontendAttrs["context:"+k] = v.Path
 			continue
 		}
+
+		// handle OCI layout
+		if strings.HasPrefix(v.Path, "oci-layout://") {
+			pathAlone := strings.TrimPrefix(v.Path, "oci-layout://")
+			// need to add "dummy.io", because reference.Parse*() funcs require we do not start with a '/'
+			named, err := reference.ParseNormalizedNamed(fmt.Sprintf("dummy.io%s", pathAlone))
+			if err != nil {
+				return nil, errors.Wrapf(err, "invalid oci-layout context %s, must be oci-layout:///path/to/layout@sha256:hash", v.Path)
+			}
+			// check that we had a digest
+			digested, ok := named.(reference.Digested)
+			if !ok {
+				return nil, errors.Wrapf(err, "invalid oci-layout context %s, must be oci-layout:///path/to/layout@sha256:hash", v.Path)
+			}
+			// now ensure that we start with a `/`
+			localPath := fmt.Sprintf("/%s", reference.Path(named))
+			store, err := local.NewStore(localPath)
+			if err != nil {
+				return nil, errors.Wrapf(err, "invalid store at %s", localPath)
+			}
+			// now we can add it
+			target.OCIStores = map[string]content.Store{
+				k: store,
+			}
+			target.FrontendAttrs["context:"+k] = "oci-layout:" + k
```

Comment:
The value in here nees to include digest that you are expecting in https://github.com/moby/buildkit/blob/master/frontend/dockerfile/builder/build.go#L892

Reply:
Ah, that is it. I went through our buildkit PR, and could not recall which part did this. That helps.

Reply:
Take a look, I _think_ I got it right.